packages/api-client/src/executions.ts

```diff
@@ -13,16 +13,23 @@ const { invokeApi } = require('./cumulusApiClient');
  *                                     to cumulusApiClient.invokeApi
  * @returns {Promise<Object>}        - the execution fetched by the API
  */
-const getExecution = async ({ prefix, arn, callback = invokeApi }) => {
-  const result = await callback({
+export const getExecution = async (params: {
+  prefix: string,
+  arn: string,
+  callback?: InvokeApiFunction
+}) => {
+  const { prefix, arn, callback = invokeApi } = params;
+
+  const response = await callback({
     prefix,
     payload: {
       httpMethod: 'GET',
       resource: '/{proxy+}',
       path: `/executions/${arn}`
     }
-  }).then(({ body }) => JSON.parse(body));
-  return result;
+  });
+
+  return <ExecutionRecord>JSON.parse(response.body);
```

Comment:
todo: followup on conversation re: explicit vs inferred return types and multi-type function returns from slack

Reply:
After our discussion in Slack, I'm going to go through and make the return types explicit rather than doing the type casting here.